src/app/routes/api/v3/forms/public-forms.submissions.routes.ts

```diff
@@ -67,8 +69,63 @@ PublicFormsSubmissionsRouter.route(
 )
 
 /**
+ * Submit a form response before public key encryption, performs pre-encryption
+ * steps (e.g. field validation, virus scanning) and stores the encrypted contents.
+ * @route POST /forms/:formId/submissions/storage
+ * @param response.body.required - contains the entire form submission
+ * @param captchaResponse.query - contains the reCAPTCHA response artifact, if any
+ * @returns 200 - submission made
+ * @returns 400 - submission has bad data and could not be processed
+ */
+PublicFormsSubmissionsRouter.route(
+  '/:formId([a-fA-F0-9]{24})/submissions/multirespondent',
+).post(
+  limitRate({ max: rateLimitConfig.submissions }),
+  MultirespondentSubmissionController.handleMultirespondentSubmission,
+)
+
+/**
+ * Retrieve actual response for a multirespondent mode form
+ * @route GET /forms/:formId/submissions/:submissionId
+ * @returns 200 with encrypted submission data response
+ * @returns 400 when form is not an encrypt mode form
+ * @returns 400 when Joi validation fails
+ * @returns 404 when submissionId cannot be found in the database
+ * @returns 404 when form cannot be found
+ * @returns 410 when form is archived
+ * @returns 500 when any errors occurs in database query or generating signed URL
+ */
+PublicFormsSubmissionsRouter.route(
+  '/:formId([a-fA-F0-9]{24})/submissions/:submissionId([a-fA-F0-9]{24})',
+)
+  .get(
+    MultirespondentSubmissionController.handleGetMultirespondentSubmissionForRespondent,
+  )
+  .put(
+    limitRate({ max: rateLimitConfig.submissions }),
+    MultirespondentSubmissionController.handleUpdateMultirespondentSubmission,
+  )
+
+/**
+ * Get S3 presigned post data for attachments in a submission.
+ * @route POST /forms/:formId/submissions/get-s3-presigned-post-data
+ * @param response.body.required - contains field ids and sizes of attachments
+ * @returns 200 - presigned post data generated
+ * @returns 400 - ids are invalid or attachment size exceeds limit
+ * @returns 500 - failed to generate presigned post data
+ */
+PublicFormsSubmissionsRouter.route(
+  '/:formId([a-fA-F0-9]{24})/submissions/get-s3-presigned-post-data',
+).post(
+  limitRate({ max: rateLimitConfig.submissions }),
+  SubmissionController.handleGetS3PresignedPostData,
```

Comment:
Remember that we discussed that this change is backwards incompatible. What was our rollout strategy again?